### PR TITLE
Add more methods and unit tests

### PR DESCRIPTION
**Pull request by KPreisner**
# Add more methods and unit tests

This PR introduces new python SDK methods that implement interacting with more of the Sindri API endpoints. NOTE: All of these API endpoints are currently hidden on https://sindri.app/docs/reference/api/circuits/
- delete circuit
- delete proof
- get team detail

This PR introduces a *first draft* of pytest unit tests. #23 outlines a future update to these unit tests.

### Reviewer testing
To run the new unit tests, which also test the new methods. See `tests/README.md`.
- [ ] Run unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,3 +17,6 @@ Documentation for this repo is automatically generated from the Python docstring
 [![](https://img.shields.io/github/license/sindri-labs/sindri-python?style=for-the-badge)](https://img.shields.io/github/license/sindri-labs/sindri-python?style=for-the-badge)
 
 `sindri-python` is licensed under a [MIT License](LICENSE) and is copyright [Sindri Labs, Inc.](https://sindri.app).
+
+## Unit Tests
+See `tests/README.md` for more information.
```
Comment (sangaline) [inline README.md]:
```suggestion
See [`tests/README.md`](tests/README.md) for more information.
```

File: src/sindri/sindri.py [hunk 2]
```diff
@@ -123,12 +123,12 @@ def _hit_api(self, method: str, path: str, data=None, files=None) -> tuple[int,
         try:
             if method == "POST":
                 response = requests.post(
-                    full_path, headers=self.headers_json, data=data, files=files
+                    full_path, headers=self.headers_json, data=data, files=files, verify=False
```
Comment (sangaline) [inline src/sindri/sindri.py]: What's the purpose of `verify=False` here? Isn't this opening people up to MITM attacks?

File: src/sindri/sindri.py [hunk 3]
```diff
@@ -353,6 +353,41 @@ def create_circuit(self, circuit_upload_path: str, wait: bool = True) -> str:
...
+
+    def delete_proof(
+        self, proof_id: str
+    ) -> tuple[int, dict | list]:
+        """
+        Delete a proof by hitting the `/proof/<proof_id>/delete` endpoint.
+
+        Return the `response.status_code` and the JSON decoded `response`.
+        This may raise `Sindri.APIError`.
+        This does not check if the `response.status_code` is the successful response (200).
```
Comment (sangaline) [inline src/sindri/sindri.py]: It does though? Same comment for `delete_circuit` above.

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+# Install necessary modules
+pip install requests pytest
+# Run `deactivate` to deactivate virtual environment
+```
+
+### Environment Variables
+When running `pytest`, these environment variables must be set in your environment. You may prefix your command with the environment variables as well. For example: `SINDRI_API_KEY=your_api_key pytest`
+
+- `SINDRI_API_KEY=your_api_key` *Required*
+- `SINDRI_API_URL=https://sindri.app/api` *Optional. Default value shown*
```
Comment (sangaline) [inline tests/README.md:30]: This predates this PR, but we should probably make these environment variables consistent across the SDKs. The TS SDK has been using `SINDRI_BASE_URL` without the `/api` suffix.

File: tests/test_sindri.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import os
+import time
+
+from src.sindri.sindri import Sindri
+
+
+# Load sample data.
+# This assumes the sindri-resources repo is cloned in the parent dir of this repo
```
Comment (sangaline) [inline tests/test_sindri.py:8]: Might want to use a git submodule if the tests depend on it?

File: tests/test_sindri.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+assert os.path.exists(circuit_dir)
+assert os.path.exists(proof_input_file_path)
+proof_input = ""
+with open(proof_input_file_path, "r") as f:
+    proof_input = f.read()
+
+
+# Create an instance of the Sindri SDK
+api_key = os.environ.get("SINDRI_API_KEY", "unset")
+api_url = os.environ.get("SINDRI_API_URL", "https://sindri.app/api")
```
Comment (sangaline) [inline tests/test_sindri.py:24]: With API tests, I think you'd typically use something like [vcrpy](https://vcrpy.readthedocs.io/en/latest/) to record responses, then have separate test commands for recording vs playing back. I'll buy it being out of scope for this PR, but doing something like that and running tests on CI is probably the direction we want to head with this.

File: tests/README.md (hunk 4 above)
Comment (KPreisner) [reply tests/README.md:30]: We should make a separate ticket for this then. We'll have to update other repos that depend on this ENV.

File: tests/test_sindri.py (hunk 6 above)
Comment (KPreisner) [reply tests/test_sindri.py:24]: I created a new issue for updating the unit tests with recorded responses: #23

File: src/sindri/sindri.py (hunk 3 above)
Comment (KPreisner) [reply src/sindri/sindri.py]: Updated docstring: [4b6f5ef](https://github.com/Sindri-Labs/sindri-python/pull/21/commits/4b6f5ef85b5d0a339d92107ad5e560f5e70656f2)

File: src/sindri/sindri.py (hunk 2 above)
Comment (KPreisner) [reply src/sindri/sindri.py]: Great point. It was to make Sindri employee development a little easier if they stood up an instance of the Sindri API with fake certs. However, the security hole is not worth it.. removed in [4b6f5ef](https://github.com/Sindri-Labs/sindri-python/pull/21/commits/4b6f5ef85b5d0a339d92107ad5e560f5e70656f2)

File: tests/test_sindri.py (hunk 5 above)
Comment (KPreisner) [reply tests/test_sindri.py:8]: I think we can hold off with the git submodule if we plan on recording API responses in #23

**Review by sangaline — APPROVED** [review]
lgtm

**Pull request merged**